Trainings #03-1 #03-2 #03-3

krasnovsasha opened the pull request:
#03-1 method variable
#03-2 class variable
#03-3 local variable types output

krasnovsasha committed:
feat: output message has timestamped

krasnovsasha committed:
feat: counter for output messages

krasnovsasha committed:
feat: variable decoratedMessage type switched from String to var

hellenaalex commented on src/main/java/com/tcs/edu/decorator/TimestampMessageDecorator.java:
```diff
@@ -0,0 +1,21 @@
+     * @see com.tcs.edu.printer.ConsolePrinter#print(String)
+     */
+    public static String decorate(String message){
+        String decoratedMessage;
```
Тут можно объединить объявление и инициализацию переменной. А можно и вообще обойтись без переменной, сразу в return склеивать строки

hellenaalex commented on src/main/java/com/tcs/edu/decorator/TimestampMessageDecorator.java:
```diff
@@ -0,0 +1,22 @@
+
+/**
+ * The {@code TimestampMessageDecorator} class contains method {@code decorate} for change
+ * base message output.
```
Не хватает следующих комментариев:
побочные эффекты
автор

hellenaalex commented on src/main/java/com/tcs/edu/decorator/TimestampMessageDecorator.java:
```diff
@@ -0,0 +1,22 @@
+     */
+    public static String decorate(String message){
+        messageCount++;
+        var decoratedMessage = messageCount + " " + new Date() + " " + message;
```
По условиям задания нужно использовать   Instant.now() вместо Date()
И в задании, вроде бы ничего нет про счетчик.

krasnovsasha committed:
fix: updating after review
Date() changed to Instant.now()
Added @author and side effects to javadoc
Removed unnecessary variable in method decorate()
